### PR TITLE
Improved api as per code review suggestions.

### PR DESCRIPTION
**Pull request by argast**

### CODE REVIEW

File: src/main/scala/com/timeout/healthchecks/package.scala [hunk 1]
```diff
@@ -9,21 +9,20 @@ import scala.util.Try
 
 package object healthchecks {
 
-  class HealthCheck(val name: String, check: AsyncCheck) {
+  class HealthCheck(val name: String, check: => Future[CheckResult]) {
```
Comment (ostapneko) [inline src/main/scala/com/timeout/healthchecks/package.scala:12]: ~~Don't need to have a by-name parameter with a `Future`~~ I'm wrong

File: src/main/scala/com/timeout/healthchecks/package.scala [hunk 2]
```diff
@@ -9,21 +9,20 @@ import scala.util.Try
...
-      check().recover {
+      check.recover {
         case t: Throwable => t.getMessage.invalidNel[Unit]
       }.map(_.leftMap(r => r.map(addNameToErrorMessage)))
     }
   }
 
-  def healthCheck(name: String, c: Check): HealthCheck = new HealthCheck(name, () => Future.fromTry(Try(c())))
-  def asyncHealthCheck(name: String, c: AsyncCheck): HealthCheck = new HealthCheck(name, c)
+  def healthCheck(name: String)(c: => ValidatedNel[String, Unit]): HealthCheck = new HealthCheck(name, Future.fromTry(Try(c)))
```
Comment (ostapneko) [inline src/main/scala/com/timeout/healthchecks/package.scala]: You could reuse your `CheckResult` type

**Comment by ostapneko** [comment]
I think the API is a bit better overall

File: src/main/scala/com/timeout/healthchecks/package.scala (hunk 1 above)
Comment (argast) [inline src/main/scala/com/timeout/healthchecks/package.scala:12]: We need it to delay evaluation when healthcheck is actually run,

File: src/main/scala/com/timeout/healthchecks/package.scala (hunk 2 above)
Comment (argast) [inline src/main/scala/com/timeout/healthchecks/package.scala]: yeah, I missed few spots. Fixed.

**Comment by ostapneko** [comment]
👍

**Pull request merged**